modular_skyrat/modules/dirtening/code/dirtening.dm

```diff
@@ -0,0 +1,48 @@
+/datum/component/dirtening
+	///the probability that the component will do its dirty work
+	var/dirtening_chance = 10
+	///the amount that will change on the alpha of the dirt
+	var/dirtening_amount = 5
+	///the parent/linked human for the component
+	var/mob/living/carbon/human/linked_human
+
+/datum/component/dirtening/Initialize(chance = null, amount = null)
+	if(!ishuman(parent))
+		return COMPONENT_INCOMPATIBLE
+	linked_human = parent
+
+	if(chance)
+		dirtening_chance = chance
+
+	if(amount)
+		dirtening_amount = amount
+
+	RegisterSignal(linked_human, COMSIG_MOVABLE_MOVED, .proc/try_dirtening)
+
+/datum/component/dirtening/proc/try_dirtening()
+	var/turf/human_turf = get_turf(linked_human)
+	//we cant dirty space turfs or closed turfs
+	if(isspaceturf(human_turf) || isclosedturf(human_turf))
+		return
+	//if we aren't lucky
+	if(!prob(dirtening_chance))
+		return
+	//try to find the dirt on the current turf
+	var/obj/effect/decal/cleanable/dirt/chosen_dirt = locate() in human_turf
+	//if there is no dirt, add it and the dirty amount
+	if(!chosen_dirt)
+		chosen_dirt = new(human_turf)
```

Comment:
I'm not sure this does what you want it to do, I feel like this will just create more of the turf the person was on.

Reply:
My bad I'm being an idiot.

Reply:
resolved.